### PR TITLE
Fix Outlook follow-up draft creation loop

### PR DESCRIPTION
**Pull request by elie222**
# User description
## Summary

Fixes cascading draft creation in follow-up reminders for Outlook users. The root cause was that `getLatestMessageInThread` returned newly-created drafts as the latest message, causing the same thread to be reprocessed on each cron cycle and generating duplicate drafts.

## Changes

- Filter draft messages in `getLatestMessageInThread` and `getThreadsWithLabel` (Outlook provider)
- Replace upsert-by-compound-key with findFirst/update pattern to reuse tracker rows across messageId changes  
- Clean up orphaned drafts if tracker update fails
- Update tests to verify tracker reuse behavior

This prevents the ~1,500 duplicate drafts that accumulated for the affected user and provides defense-in-depth guards against similar issues.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
```mermaid
graph LR
processFollowUpsForType_("processFollowUpsForType"):::modified
PRISMA_DB_("PRISMA_DB"):::modified
PRISMA_HELPERS_("PRISMA_HELPERS"):::added
generateFollowUpDraft_("generateFollowUpDraft"):::modified
EMAIL_PROVIDER_("EMAIL_PROVIDER"):::modified
SENTRY_("SENTRY"):::modified
processFollowUpsForType_ -- "Now queries latest unresolved tracker before upsert/create." --> PRISMA_DB_
processFollowUpsForType_ -- "Now updates tracker with messageId, sentAt, followUpAppliedAt." --> PRISMA_DB_
processFollowUpsForType_ -- "Now creates tracker with emailAccountId, threadId, messageId, timestamps." --> PRISMA_DB_
processFollowUpsForType_ -- "Handles duplicate-create by updating resolved flag and timestamps." --> PRISMA_DB_
processFollowUpsForType_ -- "Adds isDuplicateError check to handle unique constraint errors." --> PRISMA_HELPERS_
generateFollowUpDraft_ -- "Deletes orphaned provider draft when tracker update fails." --> EMAIL_PROVIDER_
generateFollowUpDraft_ -- "Logs and captures exceptions when tracker update fails." --> SENTRY_
classDef added stroke:#15AA7A
classDef removed stroke:#CD5270
classDef modified stroke:#EDAC4C
linkStyle default stroke:#CBD5E1,font-size:13px
```

Prevents infinite draft creation loops in Outlook by filtering out draft messages when identifying the latest message in a thread. Enhances the thread tracking logic to reuse existing tracker records and ensures orphaned drafts are cleaned up if database updates fail.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1608?tool=ast&topic=Outlook+Provider+Fix>Outlook Provider Fix</a>
        </td><td>Filter out draft messages in <code>OutlookProvider</code> within <code>getThreadsWithLabel</code> and <code>getLatestMessageInThread</code> to prevent them from being processed as new thread activity.<details><summary>Modified files (1)</summary><ul><li>apps/web/utils/email/microsoft.ts</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>Reduce-Outlook-webhook...</td><td>February 15, 2026</td></tr>
<tr><td>josh@jshwrnr.com</td><td>Fix-auto-filer-reply-s...</td><td>February 10, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1608?tool=ast&topic=Tracker+Management>Tracker Management</a>
        </td><td>Replace <code>upsert</code> with a <code>findFirst</code>/<code>update</code> pattern in the follow-up processing logic to maintain tracker continuity and add error handling to delete orphaned drafts in <code>generateFollowUpDraft</code>.<details><summary>Modified files (3)</summary><ul><li>apps/web/app/api/follow-up-reminders/process.test.ts</li>
<li>apps/web/app/api/follow-up-reminders/process.ts</li>
<li>apps/web/utils/follow-up/generate-draft.ts</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>Fix-recurring-auto-dra...</td><td>February 13, 2026</td></tr>
<tr><td>joshwerner001@gmail.com</td><td>Fix-draft-generation-f...</td><td>January 22, 2026</td></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1608?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/app/api/follow-up-reminders/process.ts [hunk 1]
```diff
@@ -280,27 +280,31 @@ async function processFollowUpsForType({
...
+          resolved: false,
         },
+        orderBy: { createdAt: "desc" },
       });
 
+      const tracker = existingTracker
+        ? await prisma.threadTracker.update({
+            where: { id: existingTracker.id },
+            data: { messageId: lastMessage.id, followUpAppliedAt: now },
+          })
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/api/follow-up-reminders/process.ts]: Updated. The existing-tracker reuse path now keeps thread metadata current by filtering reuse to the current tracker type and updating `sentAt` along with `messageId` and `followUpAppliedAt`.

File: apps/web/app/api/follow-up-reminders/process.ts [hunk 2]
```diff
@@ -280,27 +281,56 @@ async function processFollowUpsForType({
...
+      let tracker;
+      if (existingTracker) {
+        tracker = await prisma.threadTracker.update({
+          where: { id: existingTracker.id },
+          data: {
+            messageId: lastMessage.id,
+            sentAt: messageDate,
+            followUpAppliedAt: now,
+          },
+        });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/api/follow-up-reminders/process.ts]: Fixed in this branch. The `existingTracker` update path now catches duplicate-key (`P2002`) and falls back to updating the conflicting composite-key row instead of aborting thread processing. I also added a regression test for duplicate-on-existing-update, and kept duplicate-on-create fallback covered.

File: apps/web/app/api/follow-up-reminders/process.ts [hunk 3]
```diff
@@ -280,27 +280,31 @@ async function processFollowUpsForType({
...
-          followUpAppliedAt: now,
-        },
-        create: {
           emailAccountId: emailAccount.id,
           threadId: thread.id,
-          messageId: lastMessage.id,
-          type: trackerType,
-          sentAt: messageDate,
-          followUpAppliedAt: now,
+          resolved: false,
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/api/follow-up-reminders/process.ts:289]: Addressed. Existing tracker lookup is now scoped by `type: trackerType` (plus unresolved), so awaiting-reply and needs-reply flows no longer reuse each others rows.
